mobile/src/main/java/org/openhab/habdroid/background/ForegroundBroadcastReceiver.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.background
+
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.IBinder
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import org.openhab.habdroid.R
+import org.openhab.habdroid.util.PrefKeys
+import org.openhab.habdroid.util.getPrefs
+import org.openhab.habdroid.util.isItemUpdatePrefEnabled
+
+class ForegroundBroadcastReceiver : Service() {
+    private var isServiceStarted = false
+    private var backgroundTasksManager = BackgroundTasksManager()
+    private var isRegistered = false
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        Log.d(TAG, "onStartCommand())")
+        when (intent?.action) {
+            ACTION_START -> startService()
+            ACTION_STOP -> stopService()
```

Comment:
This logic seems to be unneeded. I'd instead suggest to
- Call `context.stopService()` in `startOrStopService`
- Drop the actions completely
- Check whether there are matching intent filter actions unconditionally in onStartCommand() and call `stopSelf(startId)` if there aren't any
- Do receiver unregistering (if needed) in `onDestroy`

Reply:
Thanks, that looks way better.